tests/benchmarks/tpch/test_pyspark.py

```diff
@@ -0,0 +1,229 @@
+import asyncio
+import logging
+import os
+import pathlib
+import re
+import shlex
+import subprocess
+import sys
+
+import pyspark
+from conda.cli.python_api import Commands, run_command
+from distributed.diagnostics.plugin import SchedulerPlugin, WorkerPlugin
+from distributed.utils import get_ip
+
+from tests.benchmarks.tpch.conftest import DATASETS, ENABLED_DATASET  # noqa: F401
+
+from . import pyspark_queries as queries
+
+# pyspark/hadoop/aws-java-* are extra sensitive to version changes it seems.
+# ref: https://hadoop.apache.org/docs/r3.3.4/hadoop-aws/dependency-analysis.html
+# for verifying aws-java-sdk-bundle version w/ hadoop version.
+assert pyspark.__version__ == "3.4.1"
+HADOOP_AWS_VERSION = "3.3.4"  # sc._jvm.org.apache.hadoop.util.VersionInfo.getVersion()
+AWS_JAVA_SDK_BUNDLE_VERSION = "1.12.262"
+
+
+logger = logging.getLogger("coiled")
+
+PACKAGES = (
+    f"org.apache.hadoop:hadoop-client:{HADOOP_AWS_VERSION}",
+    f"org.apache.hadoop:hadoop-common:{HADOOP_AWS_VERSION}",
+    f"org.apache.hadoop:hadoop-aws:{HADOOP_AWS_VERSION}",
+    f"com.amazonaws:aws-java-sdk-bundle:{AWS_JAVA_SDK_BUNDLE_VERSION}",
+)
+
+
+def test_query_1(tpch_pyspark_client):
+    return run_tpch_pyspark(tpch_pyspark_client, queries.q1)
+
+
+def test_query_2(tpch_pyspark_client):
+    return run_tpch_pyspark(tpch_pyspark_client, queries.q2)
+
+
+def test_query_3(tpch_pyspark_client):
+    return run_tpch_pyspark(tpch_pyspark_client, queries.q3)
+
+
+def test_query_4(tpch_pyspark_client):
+    return run_tpch_pyspark(tpch_pyspark_client, queries.q4)
+
+
+def test_query_5(tpch_pyspark_client):
+    return run_tpch_pyspark(tpch_pyspark_client, queries.q5)
+
+
+def test_query_6(tpch_pyspark_client):
+    return run_tpch_pyspark(tpch_pyspark_client, queries.q6)
+
+
+def test_query_7(tpch_pyspark_client):
+    return run_tpch_pyspark(tpch_pyspark_client, queries.q7)
+
+
+def make_pyspark_submit_args(spark_master):
+    return f"""
+        --packages {','.join(PACKAGES)}
+        --exclude-packages com.google.guava:guava
+        --master=spark://{spark_master}
+        pyspark-shell
+    """
+
+
+def fix_timestamp_ns_columns(query):
+    """
+    scale100 stores l_shipdate/o_orderdate as timestamp[us]
+    scale1000 stores l_shipdate/o_orderdate as timestamp[ns] which gives:
+        Illegal Parquet type: INT64 (TIMESTAMP(NANOS,true))
+    so we set spark.sql.legacy.parquet.nanosAsLong and then convert to timestamp.
+    """
+    for name in ("l_shipdate", "o_orderdate"):
+        query = re.sub(rf"\b{name}\b", f"to_timestamp(cast({name} as string))", query)
+    return query
```

Comment:
@milesgranger please also open an issue for this. I don't feel comfortable with this string casting when we're comparing us to spark. If that means we have to regenerate the dataset, that's unfortunate but I wouldn't want us to bias spark too severely. At the very least we should confirm on smaller / other data what the impact here is and whether it can be ignored.